lucille-core/src/main/java/com/kmwllc/lucille/stage/StringHashSet.java

```diff
@@ -0,0 +1,71 @@
+package com.kmwllc.lucille.stage;
+
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.util.FileUtils;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashSet;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StringHashSet implements LookupCollection<String> {
```

Comment:
it's confusing to have a class named StringHashSet that is not a kind of HashSet but something else like LookupCollection<String>

It looks to me like we could use a regular java.util.HashSet for our purposes here
we simply need some code to properly populate the HashSet
that code could be put in a utility method inside the stage
I'm not sure I'm seeing a justification for why we need the LookupCollection interface